rmgpy/data/kinetics/rules.py

```diff
@@ -596,82 +592,143 @@ def estimateKinetics(self, template, degeneracy=1):
-                    kinetics.comment = 'Average of [{0}]'.format(
-                         ' + '.join(k.comment if k.comment != '' else ';'.join(g.label for g in t) for k, t in kineticsList))
-                    kinetics.comment +='\n'
-                    # Append standard portion of kinetics comments that appear in non-verbose mode.
-                    kinetics.comment += 'Estimated using average of templates {0}'.format(
+                    kinetics.comment += 'Exact match found' 
+            else:
+                #Using a more general node to estimate original template
+                kinetics.comment +='Estimated using template ' + matchedLeaves
+                            
+        else:
+            # We found one or more results! Let's average them together
+            kinetics = self.__getAverageKinetics([k for k, t in kineticsList])
+            # Unlike in the case of a single rule, the verbose comments for averaging are lost unless they are 
+            # appended in the following lines.  Verbose comments are filtered out in 
+            # rmgpy.rmg.model.CoreEdgeReactionModel.generateKinetics
+            kinetics.comment = 'Average of [{0}]'.format(
+                    ' + '.join(k.comment if k.comment != '' else ';'.join(g.label for g in t) for k, t in kineticsList))
+            kinetics.comment +='\n'
+            # Append standard portion of kinetics comments that appear in non-verbose mode.
+            kinetics.comment += 'Estimated using average of templates {0}'.format(
                         ' + '.join([getTemplateLabel(t) for k, t in kineticsList]),
                     )
                 
-                kinetics.comment +=  ' for rate rule ' + originalLeaves
-                kinetics.A.value_si *= degeneracy
-                if degeneracy > 1:
-                    kinetics.comment += "\n"
-                    kinetics.comment += "Multiplied by reaction path degeneracy {0}".format(degeneracy)
+        kinetics.comment += ' for rate rule ' + originalLeaves
+        kinetics.comment += ' Euclidian distance = {}.'.format(minNorm)
+        kinetics.A.value_si *= degeneracy
+        if degeneracy > 1:
+            kinetics.comment += "\n"
+            kinetics.comment += "Multiplied by reaction path degeneracy {0}".format(degeneracy)
+
+        return kinetics, (entry if 'Exact' in kinetics.comment else None)
 
-                return kinetics, entry if 'Exact' in kinetics.comment else None
+def removeIdenticalKinetics(kList):
+    """
+    removes all identical kinetics entries in kList
+    takes in a list of kinetics entries
+    returns the list with the identical kinetics entries removed
+    
+    does this based on strings, which should be fine for this specifically, since we shouldn't have any
+    identical kinetics entries in the families and all of the identical kinetics should look exactly the same
+    """
+    outSet = set()
+    outList = []
```

Comment:
We need outList because outList has the actual object outSet only has the strings